### DISCUSSION TITLE
Adding analysis pipeline

### DISCUSSION THREAD

**Discussion by hbk008** [NeuroTechX/EEG-ExPy#207] [Ideas]
These would be some general preprocessing steps which we could plan on implementing:

- Importing the raw data.
- Downsampling the data (if we would like to do connectivity analysis). 
- Removing power line noise.
- Band pass filtering.
- Re-referencing.
- Rejecting noisy channels.
- Data epoching.
- Rejecting bad epochs.
- Run ICA and remove components with artifacts.

We can add the power spectral plots and time-frequency domain analysis. For connectivity analysis and source reconstruction we can open a separate thread in the meanwhile for discussion while I'll be adding the preprocessing pipeline. Eventually, we could generate an analysis report (pdf/html) like we are doing currently with some plots like spectrograms and plots of how data is changing after different steps. Please put your thoughts on the preprocessing pipeline. I am currently also looking into AutoReject and PREP pipeline which are kind of automated versions of EEG preprocessing.

**Comment by JohnGriffiths** [comment]
Something to consider:

https://mne.tools/mne-bids-pipeline/index.html
